controllers/setup.go

```diff
@@ -36,64 +37,75 @@ func CreateAndStartReconciler(ctx context.Context, mgr controllerruntime.Manager
 		node_labeller.New(),
 	}
 
-	var requiredCrds []string
-	for i := range sspOperands {
-		requiredCrds = append(requiredCrds, sspOperands[i].RequiredCrds()...)
-	}
-
 	mgrCtx, cancel := context.WithCancel(ctx)
 	defer cancel()
+	mgrCtx = logr.NewContext(mgrCtx, mgr.GetLogger())
 
-	crdWatch := crd_watch.New(requiredCrds...)
-	// Cleanly stops the manager and exit. The pod will be restarted.
-	crdWatch.AllCrdsAddedHandler = cancel
-	crdWatch.SomeCrdRemovedHandler = cancel
-
-	err = crdWatch.Init(mgrCtx, mgr.GetAPIReader())
+	runningOnOpenShift, err := common.RunningOnOpenshift(ctx, mgr.GetAPIReader())
 	if err != nil {
 		return err
 	}
 
-	if missingCrds := crdWatch.MissingCrds(); len(missingCrds) > 0 {
-		mgr.GetLogger().Error(nil, "Some required crds are missing. The operator will not create any new resources.",
-			"missingCrds", missingCrds,
-		)
-	}
+	if runningOnOpenShift {
+		var requiredCrds []string
```

Comment:
Maybe it makes sense to move the body of the if clause to its own function?

Reply:
Thanks @0xFelix - done